flink-streaming-java/src/main/java/org/apache/flink/streaming/api/functions/sink/PrintSinkFunction.java

```diff
@@ -33,7 +33,10 @@
  * no {@code sinkIdentifier} provided, parallelism == 1
  *
  * @param <IN> Input record type
+ * @deprecated This interface will be removed in future versions. Use the new {@link PrintSink}
+ *     interface instead.
```

Comment:
I've doubled checked: `PrintSink` implements `Sink` interface which is going to be deprecated in this PR
So seems we can't propose to use it.

and may be mark `PrintSink` as well deprecated

Reply:
Verified that `PrintSink` actually uses `Sink2` and not `Sink`.